File: src/main/java/compiler/block/primitives/CharacterBlock.java
package compiler.block.primitives;

import compiler.block.Block;

public class CharacterBlock extends Block {

    private String type = "char";
    private String value;
    private String name;

    public CharacterBlock(Block superBlock, String name, String value) {
        super(superBlock, false, true);
        this.name = name;
        this.value = value;
    }

    @Override
    public void init() {

    }

    public String getType() {
        return type;
    }

    @Override
    public String getOpeningCode() {
        return "";
    }

    @Override
    public String getClosingCode() {
        return "";
    }

    @Override
    public String getBodyCode() {
        return "char "+name+" = '"+value+"';";
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: src/main/scala/compiler/block/primitives/CharacterBlock.scala
package compiler.block.primitives

import compiler.block.Block

class CharacterBlock( superBlockInit: Block, declaration : Boolean, name: String,  value: String) extends Block(superBlockInit, false, true) {

  override def init() {}

  override def getName: String = name

  override def getValue: String = value

  override def getType: String = "char"


  override def getOpeningCode: String = {
    return "char " + name + " = '" + value + "';"
  }

  override  def getClosingCode: String = {
    return ""
  }

  override def toString: String = "char: " + name + " = " + value

}